tesseract-ocr: update to 5.3.3

chrysos349 opened the pull request:
@newbluemoon
`ccextractor` was updated to the latest version, because it had to be rebuilt for tesseract-ocr-5.3.2 anyway.

@Piraty 
`arcan` was revbumped for tesseract-ocr-5.3.2 .

#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures (if supported. mark crossbuilds): 
  - i686
  - aarch64
  - armv7l
  - x86_64-musl
  - armv6l-musl
  - aarch64-musl

chrysos349 commented:
Test `boxa3_reg` failed in two cases (others are being checked at the moment). I guess I'll have to remove this test from `prog/Makefile.am` for `leptonica` to pass, or add `make_check=no` if there are more problematic tests to be found later.

P.S. All the tests were passed on my local machine.

newbluemoon commented:
Did you test if ccextractor works with `-hardsubx`? Because it segfaulted and that was for me the show stopper to update. If I have a few spare minutes I'll try and test it. Hopefully it works now :)

chrysos349 commented:
@newbluemoon 

> Did you test if ccextractor works with -hardsubx? 

Good catch! I didn't. All I did was fix compilation errors. Anyway, even after adding 'tesseract 5 support', it still didn't work:
```
Job 1, 'ccextractor -hardsubx video.mp4' terminated by signal SIGSEGV (Address boundary error)
```
`ccextractor-0.94` is too problematic for my tastes. I wasted enough time as it is on it.

So I reverted back to `ccextractor-0.93`, fixed the compilation error, and added  'tesseract 5 support'. It worked fine in my tests.

I hope my work is done here.

chrysos349 commented:
bump

Piraty commented:
by now 

`leptonica-1.82.0 -> leptonica-1.84.0`

`tesseract-ocr-4.1.1 -> tesseract-ocr-5.3.3`

Piraty commented on srcpkgs/arcan/template:
```diff
@@ -32,6 +32,12 @@ checksum="7bf083412bc61555472877313c13116431a0a36fccbf142f97559db43b4a1475
 
+case "$XBPS_TARGET_MACHINE" in
+	i686*)
+		configure_args+=" -DSSE_42_DETECTED_EXITCODE=0"
```
why?

chrysos349 replied:
not needed anymore. removed.

chrysos349 commented:
`leptonica` update to `1.84.0`
`tesseract-ocr` updated to `5.3.3`
`arcan` and `ccextractor` revbumped and rebuilt for `tesseract-5.3.3`

Piraty commented on srcpkgs/ccextractor/template:
```diff
@@ -16,7 +16,12 @@ distfiles="https://github.com/CCExtractor/${pkgname}/archive/v${version}.tar.gz"
+fi
+
 pre_configure() {
+	ln -sf libleptonica.so ${XBPS_CROSS_BASE}/usr/lib/liblept.so
```
why?

chrysos349 replied:
done
